### PR TITLE
Added FlorySchulz Distribution

### PR DESCRIPTION
**Pull request by mayankray2020**
* stats
  * Implemented FlorySchulz Distribution

### CODE REVIEW

**Comment by czgdp1807** [comment]
N.B.
```
Traceback (most recent call last):
  File "/home/runner/work/sympy/sympy/sympy/core/tests/test_args.py", line 74, in test_all_classes_are_tested
    assert not failed, "Missing classes: %s.  Please add tests for these to sympy/core/tests/test_args.py." % ", ".join(failed)
AssertionError: Missing classes: sympy.stats.drv_types.FlorySchulzDistribution.  Please add tests for these to sympy/core/tests/test_args.py
```

**Comment by mayankray2020** [comment]
@Smit-create please review this

File: sympy/stats/drv_types.py [hunk 1]
```diff
@@ -106,6 +108,78 @@ def DiscreteRV(symbol, density, set=S.Integers, **kwargs):
...
+
+    The density of the FlorySchulz distribution is given by
+
+    .. math::
+        f(k) := (p^2) k (1 - p)^{k-1}
+
+    Parameters
+    ==========
+
+    p: A probability between 0 and 1
```
Comment (czgdp1807) [inline sympy/stats/drv_types.py:147]:
```suggestion
    p:
        A real number between 0 and 1
```

File: sympy/stats/drv_types.py [hunk 2]
```diff
@@ -106,6 +108,78 @@ def DiscreteRV(symbol, density, set=S.Integers, **kwargs):
...
+    9
+
+    >>> variance(X)
+    40
+
+    References
+    ==========
+
+    https://en.wikipedia.org/wiki/Flory%E2%80%93Schulz_distribution
+
```
Comment (czgdp1807) [inline sympy/stats/drv_types.py:178]:
```suggestion
```

File: sympy/stats/drv_types.py [hunk 3]
```diff
@@ -106,6 +108,78 @@ def DiscreteRV(symbol, density, set=S.Integers, **kwargs):
...
+        p = self.p
+        return (p**2 * k * (1 - p)**(k - 1))
+
+    def _characteristic_function(self, t):
+        p = self.p
+        return p**2 * exp(I*t) / ((1 + (p - 1) * exp(I*t))**2)
+
+    def _moment_generating_function(self, t):
+        p = self.p
+        return p**2 * exp(t) / ((1 + (p- 1) * exp(t))**2)
```
Comment (czgdp1807) [inline sympy/stats/drv_types.py:132]:
```suggestion
        return p**2 * exp(t) / ((1 + (p - 1) * exp(t))**2)
```

File: sympy/stats/tests/test_discrete_rv.py [hunk 4]
```diff
@@ -41,6 +41,13 @@ def test_Poisson():
     # issue 8248
     assert x.pspace.compute_expectation(1) == 1
 
+def test_FlorySchulz():
+    p = S.One / 5
+    x = FlorySchulz('x' , p)
+    assert E(x) == 9
+    assert variance(x) == 40
+    assert density(x)(2) - 8/125 < 0.001
```
Comment (czgdp1807) [inline sympy/stats/tests/test_discrete_rv.py:49]: Please add some symbolic tests as well.

Comment (czgdp1807) [reply sympy/stats/tests/test_discrete_rv.py:49]: Tests of type in L49 should be avoided. They are quite ad-hoc in nature.

Comment (mayankray2020) [reply sympy/stats/tests/test_discrete_rv.py]: done @czgdp1807

File: sympy/stats/tests/test_discrete_rv.py [hunk 5]
```diff
@@ -41,6 +41,15 @@ def test_Poisson():
     # issue 8248
     assert x.pspace.compute_expectation(1) == 1
 
+def test_FlorySchulz():
+    a = Symbol("a")
+    z = Symbol("z")
+    x = FlorySchulz('x' , a)
+    assert E(x) == (2 - a)/a
+    assert simplify(variance(x)) == simplify(2*(1 - a)/a**2)
```
Comment (czgdp1807) [inline sympy/stats/tests/test_discrete_rv.py:49]:
```suggestion
    assert (variance(x) - 2*(1 - a)/a**2).simplify() == S(0)
```

File: sympy/stats/tests/test_discrete_rv.py [hunk 6]
```diff
@@ -41,6 +41,15 @@ def test_Poisson():
...
+    assert(density(x)(z)) == a**2*z*(1 - a)**(z - 1)
```
Comment (czgdp1807) [inline sympy/stats/tests/test_discrete_rv.py:50]:
```suggestion
    assert density(x)(z) == a**2*z*(1 - a)**(z - 1)
```

Comment (mayankray2020) [reply sympy/stats/tests/test_discrete_rv.py]: done @czgdp1807

**Comment by mayankray2020** [comment]
please review @czgdp1807

**Comment by czgdp1807** [comment]
LGTM. Will merge this tomorrow if no objections raised.

**Pull request merged**